adaptive_memory_v4.0.py

```
@@ -0,0 +1,1551 @@
+
+class EmbeddingManager:
+    """Manages embedding generation, caching, and persistence."""
+
+    def __init__(self, get_valves: Callable[[], Any], error_manager: ErrorManager):
+        self.get_valves = get_valves
+        self.error_manager = error_manager
+        self.cache: Dict[str, np.ndarray] = {}
+        self.provider: Optional[EmbeddingProvider] = None
+        self._current_provider_type = None
+
+    def _ensure_provider(self):
```

Ruff (0.14.10)

Missing return type annotation for private function `_ensure_provider`

Add return type annotation: `None`

(ANN202)